contracts/strategies/stkCvxFXS/harvester/Harvester.sol

```diff
@@ -0,0 +1,170 @@
+
+    /// @notice Change the default swap option for eth -> fxs
+    /// @param _newOption - the new option to use
+    function setSwapOption(SwapOption _newOption) external onlyOwner {
+        SwapOption _oldOption = swapOption;
+        swapOption = _newOption;
+        emit OptionChanged(_oldOption, swapOption);
+    }
+
+    /// @notice Turns oracle on or off for swap
+    function switchOracle() external onlyOwner {
+        useOracle = !useOracle;
+    }
+
+    /// @notice Sets the contract's future owner
+    /// @param _po - pending owner's address
+    function setPendingOwner(address _po) external onlyOwner {
+        pendingOwner = _po;
+    }
+
+    /// @notice Allows a pending owner to accept ownership
+    function acceptOwnership() external {
+        require(pendingOwner == msg.sender, "only new owner");
+        owner = pendingOwner;
+        pendingOwner = address(0);
+    }
+
+    /// @notice switch the forceLock option to force harvester to lock
+    /// @dev the harvester will lock even if there is a discount if forceLock is true
+    function setForceLock() external onlyOwner {
+        forceLock = !forceLock;
+    }
+
+    /// @notice Rescue tokens wrongly sent to the contracts or claimed extra
+    /// rewards that the contract is not equipped to handle
+    /// @dev Unhandled rewards can be redirected to new harvester contract
+    function rescueToken(address _token, address _to) external onlyOwner {
+        /// Only allow to rescue non-supported tokens
+        require(_token != FXS_TOKEN && _token != CVX_TOKEN, "not allowed");
+        uint256 _balance = IERC20(_token).balanceOf(address(this));
+        IERC20(_token).safeTransfer(_to, _balance);
+    }
+
+    /// @notice Sets the range of acceptable slippage & price impact
+    function setSlippage(uint256 _slippage) external onlyOwner {
+        allowedSlippage = _slippage;
+    }
+
+    /// @notice Compute a min amount of ETH based on pool oracle for cvx
+    /// @param _amount - amount to swap
+    /// @return min acceptable amount of ETH
+    function _calcMinAmountOutCvxEth(uint256 _amount)
+        internal
+        returns (uint256)
+    {
+        uint256 _cvxEthPrice = cvxEthSwap.price_oracle();
+        uint256 _amountEthPrice = (_amount * _cvxEthPrice) / 1e18;
+        return ((_amountEthPrice * allowedSlippage) / DECIMALS);
+    }
+
+    function processRewards() external onlyStrategy returns (uint256) {
+        uint256 _cvxBalance = IERC20(CVX_TOKEN).balanceOf(address(this));
+        if (_cvxBalance > 0) {
+            uint256 _minAmountOut = useOracle
+                ? _calcMinAmountOutCvxEth(_cvxBalance)
+                : 0;
+            _cvxToEth(_cvxBalance, _minAmountOut);
+        }
+        uint256 _ethBalance = address(this).balance;
+        uint256 _harvested = 0;
+
+        if (_ethBalance > 0) {
+            _swapEthForFxs(_ethBalance, swapOption);
+        }
+
+        uint256 _fxsBalance = IERC20(FXS_TOKEN).balanceOf(address(this));
```

Comment:
This method deals with many different tokens and utilities for handling them - would it be possible to create helper methods that group together the logic for each token?